sympy/printing/mathml.py

```diff
@@ -387,67 +442,485 @@ def _print_int(self, p):
         dom_element.appendChild(self.dom.createTextNode(str(p)))
         return dom_element
 
-    def apply_patch(self):
-        # Applying the patch of xml.dom.minidom bug
-        # Date: 2011-11-18
-        # Description: http://ronrothman.com/public/leftbraned/xml-dom-minidom-\
-        #                   toprettyxml-and-silly-whitespace/#best-solution
-        # Issue: http://bugs.python.org/issue4147
-        # Patch: http://hg.python.org/cpython/rev/7262f8f276ff/
 
-        from xml.dom.minidom import Element, Text, Node, _write_data
+class MathMLPresentationPrinter(MathPrinter):
+    """Prints an expression to the Presentation MathML markup language.
 
-        def writexml(self, writer, indent="", addindent="", newl=""):
-            # indent = current indentation
-            # addindent = indentation to add to higher levels
-            # newl = newline string
-            writer.write(indent + "<" + self.tagName)
+    References: https://www.w3.org/TR/MathML2/chapter3.html
+    """
 
-            attrs = self._get_attributes()
-            a_names = list(attrs.keys())
-            a_names.sort()
+    def __init__(self, settings=None):
+        Printer.__init__(self, settings)
+        from xml.dom.minidom import Document,Text
 
-            for a_name in a_names:
-                writer.write(" %s=\"" % a_name)
-                _write_data(writer, attrs[a_name].value)
-                writer.write("\"")
-            if self.childNodes:
-                writer.write(">")
-                if (len(self.childNodes) == 1 and
-                        self.childNodes[0].nodeType == Node.TEXT_NODE):
-                    self.childNodes[0].writexml(writer, '', '', '')
-                else:
-                    writer.write(newl)
-                    for node in self.childNodes:
-                        node.writexml(
-                            writer, indent + addindent, addindent, newl)
-                    writer.write(indent)
-                writer.write("</%s>%s" % (self.tagName, newl))
+        self.dom = Document()
+
+        # Workaround to allow strings to remain unescaped
+        # Based on https://stackoverflow.com/questions/38015864/python-xml-dom-minidom-please-dont-escape-my-strings/38041194
```

Comment:
Doesn't this affect `MathMLContentPrinter` too?

Reply:
I don't think this affects `MathMLContentPrinter` as we don't need to add special characters (like integration and summation signs) ourselves. However we can add this for `MathMLContentPrinter` as well it won't make any difference.